pkg/reconciler/autoscaling/kpa/kpa_test.go

```diff
@@ -166,6 +171,11 @@ func TestReconcileAndScaleToZero(t *testing.T) {
 	const key = testNamespace + "/" + testRevision
 	const deployName = testRevision + "-deployment"
 	usualSelector := map[string]string{"a": "b"}
+	fn := activatorProbe
+	defer func() {
+		activatorProbe = fn
+	}()
+	activatorProbe = func(*asv1a1.PodAutoscaler) (bool, error) { return true, nil }
```

Comment:
Instead of messing with a global, can we simply make this a field on `Reconciler`?

Reply:
Or Scaler...

Reply:
and then it can't be mock'd :-) I wasn't sure about the file. I can move it to scaler.go, though.

Reply:
It can, it just requires plumbing :)  e.g. by making it a field we enable ourselves to `t.Parallel()` the tests, which would blow up here.  I don't feel super strongly, so LGTM.